llvm/lib/Target/AIE/AIEWawRegRewriter.cpp

```diff
@@ -198,14 +223,105 @@ AIEWawRegRewriter::getVRegWithCopies(const MachineBasicBlock &MBB) const {
     }
   }
 
-  // copy to BitVector so that lookups become very cheap
+  // Copy to BitVector so that lookups become very cheap
   BitVector VRegWithCopies(MaxVReg + 1);
   for (const unsigned RegIndex : VRegs)
     VRegWithCopies[RegIndex] = true;
 
   return VRegWithCopies;
 }
 
+SmallVector<MCPhysReg, 40> getGPRAliasRegs(const AIEBaseRegisterInfo *TRI,
+                                           const MachineFunction *MF) {
+  SmallVector<MCPhysReg, 40> GPRRegs;
+  for (auto PhysReg : *TRI->getGPRRegClass(*MF)) {
+    for (MCRegAliasIterator AI(MCRegister(PhysReg), TRI, true); AI.isValid();
+         ++AI) {
+      if (find(GPRRegs, PhysReg) == GPRRegs.end())
+        GPRRegs.push_back(PhysReg);
+    }
+  }
+  return GPRRegs;
+}
+
+void AIEWawRegRewriter::accumulateUsedRegs(const MachineBasicBlock &MBB,
+                                           std::set<MCPhysReg> &FoundRegs) {
+  for (const MachineInstr &MI : MBB) {
+    for (const MachineOperand &MO : MI.all_defs()) {
```

Comment:
`all_defs` will include implicit defs as well (mostly reserved regs). Is there any reason for this?